hoplite-core/src/main/kotlin/com/sksamuel/hoplite/ConfigFailure.kt

```diff
@@ -96,6 +96,10 @@ sealed interface ConfigFailure {
     override fun description(): String = "Could not find config file $source"
   }
 
+  data class EmptySource(val source: String) : ConfigFailure {
+    override fun description(): String = "Source $source is empty"
+  }
```

Comment:
new ConfigFailure implementation with helpful description, so the user can decide whether to fail on this error or not (through the use of the `optional` flag)